flang/lib/Lower/Bridge.cpp

```diff
@@ -1473,7 +1474,25 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   /// the global to the ModuleOp as a new uniqued symbol and initialize it with
   /// the correct value. It will be referenced on demand using `fir.addr_of`.
   void instantiateGlobal(const Fortran::lower::pft::Variable &var) {
-    llvm_unreachable("Varun: put your code here");
+    const auto &sym = var.getSymbol();
+    std::string globalName = uniqueCGIdent(sym.name().ToString());
```

Comment:
Should this symbol name be mangled before using `uniqueCGIdent` ?
Otherwise you could imagine collision:

```
subroutine foo()
  real :: x = 0.
end subroutine

subroutine bar()
  real :: x = 0.
end subroutine 
```

Reply:
`uniqueCGIdent` is another type of uniquing. Here, we want to unique the symbol to a bijective name. That is `x` should generate a unique symbol with `x` in the name.

(`uniqueCGIdent` is meant to be used only for objects that are syntactically nameless. The literal string "Hello, World!" has no name. But we want to be able to create one so we can reference that literal. That's what the codegen ident uniquing is for.)